Add reproducible builds (#506)
libcsp/libcsp: doc/INSTALL.md

cmake --install builddir --component runtime
```

## Reproducible Builds

libcsp supports Reproducible Builds. To enable it, set
`CSP_REPRODUCIBLE_BUILDS` to `1`.

Please note that, when reproducible builds are enabled,
`CSP_CMP_IDENT` does not return the compilation date and time.

When reproducible builds are enabled, both the BuildID and hash values
of generated binaries remain consistent for each build. Our
reproducible builds also support building in different directories.
Thus, different users should generate precisely the same binaries,
given the same source code and build environment.

You can learn more about reproducible builds at
https://reproducible-builds.org/.

Use the following commands for each build system:

### Waf

```shell
./waf configure --enable-reproducible-builds
```

### Meson

```shell
meson setup builddir . -Denable_reproducible_builds=true
```

> ### Camek

Reviewer: ```suggestion
### Cmake
```

Also it is pretty hard to see the difference between the level 2 and 3 titles, what do you think about using a level 4 instead?

Author: Done fixing it to "CMake", which seems to be the one the official uses.

Regarding to the heading size.  Data should represent what it is.  And, UI should be addressed by CSS from the theme we are using.

```shell

> cmake -S . -B builddir -DCSP_REPRODUCIBLE_BUILDS=1

Reviewer: I don't think this is required since it will not work. The bellow text is enought imo.

If you prefer to keep it, I think it shoudl at least be coherent with the rest of the documentation. I am not sure whether  you want to recommand `ninja` or not but if you want to have the same as the `Using CMake` header, I think it should be:
```suggestion
cmake -Gninja -B builddir -DCSP_REPRODUCIBLE_BUILDS=ON
```

Author: I really don't care what users use.  As long as it works for them, I'm happy.

But I agree that the doc should be consistent. So I'll use `-GNinja` for all.  Thanks for your suggestion.

```

Note: By default, CMake embeds the build directory in the binaries,

> resulting in non-deterministic builds. To address this, use
> `CSP_REPRODUCIBLE_BUILDS=ON` or `CMAKE_SKIP_RPATH=ON` as follows:

Reviewer: ```suggestion
resulting in non-deterministic builds. To address this, use
`CMAKE_BUILD_RPATH_USE_ORIGIN=ON` or `CMAKE_SKIP_RPATH=ON` as follows:
```

I think you meant this no?

```shell

> cmake -S . -B builddir -DCSP_REPRODUCIBLE_BUILDS=ON -DCMAKE_BUILD_RPATH_USE_ORIGIN=ON

Reviewer: ```suggestion
cmake -Gninja -B builddir -DCSP_REPRODUCIBLE_BUILDS=ON -DCMAKE_BUILD_RPATH_USE_ORIGIN=ON
```

 I think it should be coherent with the rest of the documentation and since the `-S .` is the default when not precised it can be removed.
